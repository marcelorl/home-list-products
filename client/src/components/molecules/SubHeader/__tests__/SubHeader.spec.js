import createTestContext from 'react-cosmos-test/enzyme'

import fixture from '../SubHeader.fixture'

const { mount, getWrapper } = createTestContext({ fixture })

beforeEach(mount)

describe('#Breadcrumb atom', () => {
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot()
  })

  it('should have breadcrumbs', () => {
    expect(getWrapper('.breadcrumb')).toHaveLength(2)
  })

  it('should have a title', () => {
    expect(getWrapper('h1').text()).toEqual('NOVIDADES DA MODA MASCULINA INTERNACIONAL')
  })

  it('should have a description', () => {
    expect(getWrapper('.subheader__description').text())
      .toEqual('O melhor da moda masculina atual e as peças mais recentes das marcas internacionais chegaram à ' +
        'seleção online da Farfetch. Descubra novidades da moda masculina e tendências que grandes designers para ' +
        'renovar o guarda-roupas, de Salvatore Ferragamo à Burberry.')
  })
})
