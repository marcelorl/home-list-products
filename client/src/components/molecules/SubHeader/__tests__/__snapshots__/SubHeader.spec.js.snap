// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Breadcrumb atom renders component 1`] = `
<SubHeader>
  <styled.div>
    <div
      className="sc-htpNat jQagpy"
    >
      <Container
        tag="div"
      >
        <div
          className="container"
        >
          <Breadcrumb
            items={
              Array [
                Object {
                  "label": "Início",
                  "link": "/inicio",
                },
                Object {
                  "label": "Homem",
                  "link": "/homem",
                },
                Object {
                  "label": "Novidades Internacionais",
                  "link": "/",
                },
              ]
            }
          >
            <styled.ul
              className="breadcrumb"
            >
              <ul
                className="breadcrumb sc-bdVaJa Bzosa"
              >
                <styled.li
                  key="Início"
                >
                  <li
                    className="sc-bwzfXH ihlxUm"
                  >
                    <a
                      href="/inicio"
                    >
                      Início
                    </a>
                  </li>
                </styled.li>
                <styled.li
                  key="Homem"
                >
                  <li
                    className="sc-bwzfXH ihlxUm"
                  >
                    <a
                      href="/homem"
                    >
                      Homem
                    </a>
                  </li>
                </styled.li>
                <styled.li
                  key="Novidades Internacionais"
                >
                  <li
                    className="sc-bwzfXH ihlxUm"
                  >
                    <a
                      href="/"
                    >
                      Novidades Internacionais
                    </a>
                  </li>
                </styled.li>
              </ul>
            </styled.ul>
          </Breadcrumb>
          <div>
            <h1>
              NOVIDADES DA MODA MASCULINA INTERNACIONAL
            </h1>
          </div>
          <div
            className="subheader__description"
          >
            O melhor da moda masculina atual e as peças mais recentes das marcas internacionais chegaram à seleção online da Farfetch. Descubra novidades da moda masculina e tendências que grandes designers para renovar o guarda-roupas, de 
            <a
              href="/"
            >
              Salvatore Ferragamo
            </a>
             à 
            <a
              href="/"
            >
              Burberry
            </a>
            .
          </div>
        </div>
      </Container>
    </div>
  </styled.div>
</SubHeader>
`;
